octopin/pin.py

```diff
@@ -94,13 +94,18 @@ def pin(
     ] = False,
     token: Annotated[
         str | None,
-        typer.Option(help="GitHub token to use."),
+        typer.Option(
+            help="GitHub token to use. Reads GITHUB_TOKEN from the environment if not provided.", envvar="GITHUB_TOKEN"
```

Comment:
`GH_TOKEN` is also frequently used (e.g. in the `gh` CLI)

```suggestion
            help="GitHub token to use. Reads GITHUB_TOKEN or GH_TOKEN from the environment if not provided.", envvar=["GITHUB_TOKEN", "GH_TOKEN"]
```

Reply:
@zxiiro, I'm not sure about the typer syntax. Is that ok?